airbyte/cloud/credentials.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+"""Cloud credentials configuration for Airbyte Cloud API authentication.
+
+This module provides the CloudCredentials class for managing authentication
+credentials when connecting to Airbyte Cloud, OSS, or Enterprise instances.
+
+Two authentication methods are supported (mutually exclusive):
+1. OAuth2 client credentials (client_id + client_secret)
+2. Bearer token authentication
+
+Example usage with client credentials:
+    ```python
+    from airbyte.cloud.credentials import CloudCredentials
+
+    credentials = CloudCredentials(
+        client_id="your-client-id",
+        client_secret="your-client-secret",
+    )
+    ```
+
+Example usage with bearer token:
+    ```python
+    from airbyte.cloud.credentials import CloudCredentials
+
+    credentials = CloudCredentials(
+        bearer_token="your-bearer-token",
+    )
+    ```
+
+Example using environment variables:
+    ```python
+    from airbyte.cloud.credentials import CloudCredentials
+
+    # Resolves from AIRBYTE_CLOUD_CLIENT_ID, AIRBYTE_CLOUD_CLIENT_SECRET,
+    # AIRBYTE_CLOUD_BEARER_TOKEN, and AIRBYTE_CLOUD_API_URL environment variables
+    credentials = CloudCredentials.from_env()
+    ```
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from airbyte._util import api_util
+from airbyte.cloud.auth import (
+    resolve_cloud_api_url,
+    resolve_cloud_bearer_token,
+    resolve_cloud_client_id,
+    resolve_cloud_client_secret,
+)
+from airbyte.exceptions import PyAirbyteInputError
+from airbyte.secrets.base import SecretString
+
+
+@dataclass
+class CloudCredentials:
```

Comment:
I asked for this to be called `CloudConnection`, since it is inclusive of both creds and api root.